### PR TITLE
Add scala 3 compiler settings test

### LINKED ISSUE #1192: 2.4.0: value settings is not a member of dotty.tools.dotc.Compiler
Scala 2.13.6, Java 11.0.10, Ammonite 2.4.0

```scala
interp.configureCompiler(_.settings.unchecked.value = true)
                         ^^^^^^^^^^
           value settings is not a member of dotty.tools.dotc.Compiler
```

### PR DESCRIPTION
**Pull request by carlosrogue**
Official [Compiler Flags](https://ammonite.io/#CompilerFlags) documentation do not mention how to set settings for scala 3 compiler.

Maybe adding a comment like this in 2.12?:

```scala
@ // Disabling default Scala imports

@ List(1, 2, 3) + "lol"
res0: String = "List(1, 2, 3)lol"

@ interp.configureCompiler(_.settings.noimports.value = true)

@ // interp.preConfigureCompiler(ctx => ctx.setSetting(ctx.settings.YnoImports, true)) // Dotty
```

Fixes #1192

### CODE REVIEW

**Comment by carlosrogue** [comment]
@lefou should I also add also this _hint_ comment to 2.12:

```scala
@ // interp.preConfigureCompiler(ctx => ctx.setSetting(ctx.settings.YnoImports, true)) // Dotty
```

**Comment by lefou** [comment]
> @lefou should I also add also this _hint_ comment to 2.12:
> 
> ```scala
> @ // interp.preConfigureCompiler(ctx => ctx.setSetting(ctx.settings.YnoImports, true)) // Dotty
> ```

I've no idea.

**Review by lefou — COMMENTED** [review]
This look good to me. Thank you!

Do you know why the test isn't configured for Scala 2.13? Also newer Scala 2.12 version aren't covered?

**Comment by carlosrogue** [comment]
@lefou there is a note regarding this test not working on 2.13:
```scala
      // not sure why that one doesn't pass in 2.13
      // even disabling the noimports and imports settings instead of setting noimports to false
      // doesn't seem to reinstate imports
      def sv = scala.util.Properties.versionNumberString
      // In 2.12.13, I would have expected things like
      //   interp.configureCompiler(_.settings.Wconf.tryToSet(List("any:wv", "cat=unchecked:ws")))
      // to re-instate the expected warning below, to no avail :|
```
I could check this later and verify if we can at least add newer 2.12 versions as part of other MR.

File: amm/repl/src/test/scala/ammonite/session/BuiltinTests.scala [hunk 1]
```diff
@@ -143,7 +143,7 @@ object BuiltinTests extends TestSuite {
       // In 2.12.13, I would have expected things like
       //   interp.configureCompiler(_.settings.Wconf.tryToSet(List("any:wv", "cat=unchecked:ws")))
       // to re-instate the expected warning below, to no avail :|
-      if (TestUtils.scala2_12 && sv.stripPrefix("2.12.").toInt <= 12) {
+      if (TestUtils.scala2_12 && sv.stripPrefix("2.12.").toInt <= 19) {
```
Comment (carlosrogue) [inline amm/repl/src/test/scala/ammonite/session/BuiltinTests.scala:146]: > Do you know or have an idea why the test isn't configured for Scala 2.13? Also newer Scala 2.12 version aren't covered?

@lefou after covering until `2.12.19` this was successful: `mill 'amm.repl[2.12.{13,14,15,16,17,18,19}].test.testOnly' -- 'ammonite.session.BuiltinTests.settings'`.

Comment (lefou) [reply amm/repl/src/test/scala/ammonite/session/BuiltinTests.scala:146]: Great!

**Review by lefou — APPROVED** [review]
Looks good to me. Thank you!

**Pull request merged**